sync_crawler/crawler/base_crawler.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC
+from abc import abstractmethod
+import hashlib
+
+from bs4 import BeautifulSoup
+import requests
+
+
+class BaseCrawler(ABC):
+
+    def __init__(self,
+                 title,
+                 content,
+                 category,
+                 modified_date,
+                 media,
+                 url=None,
+                 url_hash=None,
+                 content_hash=None,
+                 headers=None):
+        self.title = title
+        self.content = content
+        self.category = category
+        self.modified_date = self.get_modified_date(modified_date)
+        self.media = media
+        self.url = self.get_page(url, headers)
+        self.url_hash = url_hash if url_hash else self.generate_hash(
+            url) if url else None
+        self.content_hash = content_hash if content_hash else self.generate_hash(
+            content) if content else None
+        self.url_hash = url_hash if url_hash else self.generate_hash(
+            url) if url else None
+        self.content_hash = content_hash if content_hash else self.generate_hash(
+            content) if content else None
+
+    @abstractmethod
+    def get_page(self, url, headers):
+        try:
+            r = requests.get(url, headers)
```

Comment:
Missing timeout argument for method `requests.get` can cause your program to hang indefinitely.